### PR TITLE
lib/py: Reduce code duplication

### PR DESCRIPTION
**Pull request by behouba**
This pull request proposes approaches to reduce code duplication in `lib/py/images/images.py`.

Fixes: #340

### CODE REVIEW

File: lib/py/images/images.py [hunk 1]
```diff
@@ -68,6 +68,20 @@ class MagicException(Exception):
     def __init__(self, magic):
         self.magic = magic
 
+# A helper function to decode base64 data
+# based on the version of the python interpreter
+def decode_base64_data(data):
```
Comment (rst0git) [inline lib/py/images/images.py:73]: @behouba Would you be able to make this comment a [docstring](https://peps.python.org/pep-0257/)?
For example:
```python
def decode_base64_data(data):
    """A helper function to decode base64 data."""
```

File: lib/py/images/images.py [hunk 2]
```diff
@@ -68,6 +68,20 @@ class MagicException(Exception):
...
+    if (sys.version_info > (3, 0)):
+        return base64.decodebytes(str.encode(data))
+    else:
+        return base64.decodebytes(data)
+
+# A helper function to write base64 encoded data to a file
+def write_base64_data(f, data):
```
Comment (rst0git) [inline lib/py/images/images.py:80]: Ditto
```python
def write_base64_data(f, data):
    """A helper function to write base64 encoded data to a file."""
```

File: lib/py/images/images.py (hunk 1 above)
Comment (behouba) [reply lib/py/images/images.py:72]: Done!

File: lib/py/images/images.py (hunk 2 above)
Comment (behouba) [reply lib/py/images/images.py:80]: Done!

File: lib/py/images/images.py [hunk 3]
```diff
@@ -415,23 +411,15 @@ def skip(self, f, pbuff):
         f.seek(round_up(size, sizeof_u64), os.SEEK_CUR)
         return size
 
+    def _get_sem_array(self):
+        s = array.array('H')
+        if s.itemsize != sizeof_u16:
+            raise Exception("Array size mismatch")
+        return s
 
 class ipc_msg_queue_handler:
```
Comment (rst0git) [inline lib/py/images/images.py:420]: Would you be able to add two blank lines before every class definition?
```console
$ flake8 --select E302 --config=scripts/flake8.cfg lib/py/images/images.py
lib/py/images/images.py:373:1: E302 expected 2 blank lines, found 1
lib/py/images/images.py:387:1: E302 expected 2 blank lines, found 1
lib/py/images/images.py:420:1: E302 expected 2 blank lines, found 1
lib/py/images/images.py:466:1: E302 expected 2 blank lines, found 1
```

Comment (behouba) [reply lib/py/images/images.py:420]: Sure!

Comment (adrianreber) [reply lib/py/images/images.py:423]: @behouba Can you also add `lib/py/images/images.py` to the list of files `flake8` is run during CI. Have a look at the Makefile and add `lib/py/images/images.py` to where `flake8` is called.

**Review by adrianreber — APPROVED** [review]
Looks good. Thanks.

**Comment by adrianreber** [comment]
Please rebase.

**Review by rst0git — APPROVED** [review]
LGTM

**Pull request merged**